In [1]:
#import the packeges
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

#import the stats package
import scipy.stats as stats

### BUSINESS PROBLEM - 1

In [2]:
#upload the data
loans=pd.read_csv('LoansData.csv')

In [3]:
loans.head()

LoanID  Amount.Requested  Amount.Funded.By.Investors Interest.Rate  \
0       1           20000.0                     20000.0         8.90%   
1       2           19200.0                     19200.0        12.12%   
2       3           35000.0                     35000.0        21.98%   
3       4           10000.0                      9975.0         9.99%   
4       5           12000.0                     12000.0        11.71%   

  Loan.Length        Loan.Purpose Debt.To.Income.Ratio State Home.Ownership  \
0   36 months  debt_consolidation               14.90%    SC       MORTGAGE   
1   36 months  debt_consolidation               28.36%    TX       MORTGAGE   
2   60 months  debt_consolidation               23.81%    CA       MORTGAGE   
3   36 months  debt_consolidation               14.30%    KS       MORTGAGE   
4   36 months         credit_card               18.78%    NJ           RENT   

   Monthly.Income FICO.Range  Open.CREDIT.Lines  Revolving.CREDIT.Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   
3         3833.33    695-699               10.0                    9346.0   
4         3195.00    695-699               11.0                   14469.0   

   Inquiries.in.the.Last.6.Months Employment.Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years  
3                             0.0           5 years  
4                             0.0           9 years

In [4]:
loans.rename(columns={'Amount.Requested':'Amount_Requested','Amount.Funded.By.Investors':'Amount_Funded_By_Investors','Interest.Rate':'Interest_Rate','Loan.Length':'Loan_Length','Loan.Purpose':'Loan_Purpose','Debt.To.Income.Ratio':'Debt_To_Income_Ratio','Home.Ownership':'Home_Ownership','Monthly.Income':'Monthly_Income','FICO.Range':'FICO_Range','Open.CREDIT.Lines':'Open_CREDIT_Lines','Revolving.CREDIT.Balance':'Revolving_CREDIT_Balance','Inquiries.in.the.Last.6.Months':'Inquiries_in_the_last_6_Months','Monthly.Income':'Monthly_Income','FICO.Range':'FICO_Range','Open.CREDIT.Lines':'Open_CREDIT_Lines','Revolving.CREDIT.Balance':'Revolving_CREDIT_Balance','Employment.Length':'Employment_Length'},inplace=True)


In [5]:
loans.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 15 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   LoanID                          2500 non-null   int64  
 1   Amount_Requested                2499 non-null   float64
 2   Amount_Funded_By_Investors      2499 non-null   float64
 3   Interest_Rate                   2498 non-null   object 
 4   Loan_Length                     2500 non-null   object 
 5   Loan_Purpose                    2500 non-null   object 
 6   Debt_To_Income_Ratio            2499 non-null   object 
 7   State                           2500 non-null   object 
 8   Home_Ownership                  2499 non-null   object 
 9   Monthly_Income                  2499 non-null   float64
 10  FICO_Range                      2498 non-null   object 
 11  Open_CREDIT_Lines               2497 non-null   float64
 12  Revolving_CREDIT_Balance        24

In [6]:
#data cleaning
loans['Interest_Rate']=loans['Interest_Rate'].str.replace('%','')
loans['Interest_Rate']=loans['Interest_Rate'].astype(float)

In [7]:
loans['Loan_Length']=loans['Loan_Length'].str.replace('months','')
loans['Loan_Length']=loans['Loan_Length'].astype(float)

In [8]:
#finding and replacing missing values
loans.isna().sum()

LoanID                             0
Amount_Requested                   1
Amount_Funded_By_Investors         1
Interest_Rate                      2
Loan_Length                        0
Loan_Purpose                       0
Debt_To_Income_Ratio               1
State                              0
Home_Ownership                     1
Monthly_Income                     1
FICO_Range                         2
Open_CREDIT_Lines                  3
Revolving_CREDIT_Balance           3
Inquiries_in_the_last_6_Months     3
Employment_Length                 77
dtype: int64

In [9]:
loans['Amount_Requested']=np.where(loans['Amount_Requested'].isna(),loans['Amount_Requested'].mean(),loans['Amount_Requested'])
loans['Interest_Rate']=np.where(loans['Interest_Rate'].isna(),loans['Interest_Rate'].mean(),loans['Interest_Rate'])
loans['FICO_Range']=np.where(loans['FICO_Range'].isna(),loans['FICO_Range'].mode(),loans['FICO_Range'])
loans['Home_Ownership']=np.where(loans['Home_Ownership'].isna(),loans['Home_Ownership'].mode(),loans['Home_Ownership'])

In [10]:
loans.isna().sum()

LoanID                             0
Amount_Requested                   0
Amount_Funded_By_Investors         1
Interest_Rate                      0
Loan_Length                        0
Loan_Purpose                       0
Debt_To_Income_Ratio               1
State                              0
Home_Ownership                     0
Monthly_Income                     1
FICO_Range                         0
Open_CREDIT_Lines                  3
Revolving_CREDIT_Balance           3
Inquiries_in_the_last_6_Months     3
Employment_Length                 77
dtype: int64

In [11]:
loans.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 15 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   LoanID                          2500 non-null   int64  
 1   Amount_Requested                2500 non-null   float64
 2   Amount_Funded_By_Investors      2499 non-null   float64
 3   Interest_Rate                   2500 non-null   float64
 4   Loan_Length                     2500 non-null   float64
 5   Loan_Purpose                    2500 non-null   object 
 6   Debt_To_Income_Ratio            2499 non-null   object 
 7   State                           2500 non-null   object 
 8   Home_Ownership                  2500 non-null   object 
 9   Monthly_Income                  2499 non-null   float64
 10  FICO_Range                      2500 non-null   object 
 11  Open_CREDIT_Lines               2497 non-null   float64
 12  Revolving_CREDIT_Balance        24

# a)  Intrest rate is varied for different loan amounts (Less intrest charged for high loan amounts)


##### 1.) Define Null Hypothesis (H0):
              #H0 : Intrest rate is not varied for different loan amounts.
    
#### 2.) Define Alternate Hyothesis (Ha):
              #Ha : Intrest rate is  varied for different loan amounts.
    
#### 3.) Define CI and p-value:
              #CI = 95%
              #p-value=0.05%
        
#### 4.) Perform Test:
              #Pearsonr Coefficient Test

In [12]:
stats.pearsonr(loans['Interest_Rate'],loans['Amount_Requested'])

PearsonRResult(statistic=0.3324464059665682, pvalue=1.4148683739785434e-65)

#### 5.)Conclusion:
            #Since the pearsonr test score is 0.332 which is a direct intermediate relation
            #Hence we fail to accept null hypothesis

### b) Loan length is directly effecting intrest rate

### 1.) Define Null Hypothesis (H0):
          #H0 : Loan length is not directly effecting interest rate .
### 2.) Define Alternate Hyothesis (Ha):
          #Ha : Loan length is  directly effecting interest rate .
### 3.) Define CI and p-value:
          #CI = 95%
          #p-value=0.05%
    
### 4.) Perform Test:
          #Pearsonr Coefficient Test

In [13]:
stats.pearsonr(loans['Loan_Length'],loans['Interest_Rate'])

PearsonRResult(statistic=0.42417192416620575, pvalue=9.549596064332338e-110)

### 5.)Conclusion:
          # Since the pearsonr test score is 0.424 which is a direct intermediate relation
          # hence we fail to accept null hypothesis

### c.)  Inrest rate varies for different purpose of loans

In [14]:
loans['Loan_Purpose'].value_counts()

Loan_Purpose
debt_consolidation    1307
credit_card            444
other                  201
home_improvement       152
major_purchase         101
small_business          87
car                     50
wedding                 39
medical                 30
moving                  29
vacation                21
house                   20
educational             15
renewable_energy         4
Name: count, dtype: int64

In [15]:
debt=loans.Interest_Rate[loans['Loan_Purpose']=='debt_consolidation']
credit=loans.Interest_Rate[loans['Loan_Purpose']=='credit_card']
other=loans.Interest_Rate[loans['Loan_Purpose']=='other']
home=loans.Interest_Rate[loans['Loan_Purpose']=='home_improvement']
major=loans.Interest_Rate[loans['Loan_Purpose']=='major_purchase']
small=loans.Interest_Rate[loans['Loan_Purpose']=='small_business']
car=loans.Interest_Rate[loans['Loan_Purpose']=='car']
wedding=loans.Interest_Rate[loans['Loan_Purpose']=='wedding']
medical=loans.Interest_Rate[loans['Loan_Purpose']=='medical']
moving=loans.Interest_Rate[loans['Loan_Purpose']=='moving']
vacation=loans.Interest_Rate[loans['Loan_Purpose']=='vacation']
house=loans.Interest_Rate[loans['Loan_Purpose']=='house']
education=loans.Interest_Rate[loans['Loan_Purpose']=='educational']
energy=loans.Interest_Rate[loans['Loan_Purpose']=='renewable_energy']

In [16]:
u1=debt.mean()
u2=credit.mean()
u3=other.mean()
u4=home.mean()
u5=major.mean()
u6=small.mean()
u7=car.mean()
u8=wedding.mean()
u9=medical.mean()
u10=moving.mean()
u11=vacation.mean()
u12=house.mean()
u13=education.mean()
u14=energy.mean()
print('debt mean:',u1)
print('credit mean:',u2)
print('other mean:',u3)
print('home mean:',u4)
print('major mean:',u5)
print('small mean:',u6)
print('car mean:',u7)
print('wedding mean:',u8)
print('medical mean:',u9)
print('moving mean:',u10)
print('vacation mean:',u11)
print('house mean:',u12)
print('education mean:',u13)
print('energy mean:',u14)

debt mean: 13.58691660290742
credit mean: 13.05068781151047
other mean: 13.15955223880597
home mean: 11.594276315789473
major mean: 10.797821782178216
small mean: 12.83919540229885
car mean: 11.113000000000001
wedding mean: 12.073215084888425
medical mean: 11.649666666666667
moving mean: 13.621034482758622
vacation mean: 11.966190476190476
house mean: 13.447999999999999
education mean: 11.007333333333332
energy mean: 9.8775


### 1.) Define Null Hypothesis (H0):
      #H0 : Inrest rate does not  varies for different purpose of loans .
      #u1=u2=u3=.......=u14
### 2.) Define Alternate Hyothesis (Ha):
      #Ha : Inrest rate varies for different purpose of loans .
      #u1=!u2=!=!u3=!............=!u14
### 3.) Define CI and p-value:
      #CI = 95%
      #p-value=0.05%
### 4.) Perform Test:
      #F-Test

In [17]:
stats.f_oneway(debt,credit,other,home,major,small,car,wedding,medical,moving,vacation,house,education,energy)

F_onewayResult(statistic=7.482638784097436, pvalue=1.1599665535989115e-14)

### 5.) Conclusion:
         # new p-value = 1.159...e-14 < 0.05
         # hence we fail to accept null hyothesis

### d.)  There is relationship between FICO scores and Home Ownership. It means that, People with owning home will have  high FICO scores.


In [18]:
loans['FICO_Range']=loans['FICO_Range'].astype(str).str.split('-')

#replace ech range with range mean
loans['FICO_Range']=loans['FICO_Range'].apply(lambda x: (int(x[0])+int(x[1]))/2)

In [19]:
loans.Home_Ownership.value_counts()

Home_Ownership
MORTGAGE    1148
RENT        1146
OWN          200
OTHER          5
NONE           1
Name: count, dtype: int64

In [20]:
mor=loans.FICO_Range[loans['Home_Ownership']=='MORTGAGE']
rent=loans.FICO_Range[loans['Home_Ownership']=='RENT']
own=loans.FICO_Range[loans['Home_Ownership']=='OWN']
other=loans.FICO_Range[loans['Home_Ownership']=='OTHER']

### 1.) Define Null Hypothesis (H0):
      #H0 : FICO SCORE and Home ownership will not have a direct relationship
### 2.) Define Alternate Hyothesis (Ha):
      #Ha : FICO SCORE and Home ownership will  have a direct relationship
### 3.) Define CI and p-value:
      #CI = 95%
      #p-value=0.05%
### 4.) Perform Test:
      #F-Test

In [21]:
stats.f_oneway(mor,rent,own,other)

F_onewayResult(statistic=23.29739165105827, pvalue=7.180580645216916e-15)

### Conclusion:
       # since the new p-value is 7.180........e-15< 0.05
       # hence, we reject the null hypothesis
        

### BUSINESS PROBLEM - 2

In [22]:
price=pd.read_csv('Price_Quotes.csv')

In [23]:
price.head()

Order_Number  Barry_Price  Mary_Price
0             1          126         114
1             2          110         118
2             3          138         114
3             4          142         111
4             5          146         129

In [24]:
u1=price.Barry_Price.mean()
u2=price.Mary_Price.mean()
print('Barry mean price:', u1)
print('Mary mean price:', u2)

Barry mean price: 124.33333333333333
Mary mean price: 114.75


### BUSINESS PROBLEM: We would like to assess if there is any difference in the average 
### price quotes provided by Mary and Barry.


### 1.) Define Null Hypothesis (H0):
          #H0 : There is no difference in the average quotes provided by Barry and Mary.
                 u1=u2    
### 2.) Define Alternate Hyothesis (Ha):
          #Ha : There is  difference in the average quotes provided by Barry and Mary.
                 u1<>u2
### 3.) Define CI and p-value:
          #CI = 95%
          #p-value=0.05%
    
### 4.) Perform Test:
          #T-Test

In [25]:
stats.ttest_ind(price.Barry_Price,price.Mary_Price)

TtestResult(statistic=1.4147436739281787, pvalue=0.17114226132118285, df=22.0)

### Conclusion:
         # since new p-value is 0.17 > 0.05
         # hence we fail to reject null hypothesis

### BUSINESS PROBLEM - 3

In [26]:
treat=pd.read_csv('Treatment_Facility.csv')

In [27]:
treat.head()

Month Reengineer  Employee_Turnover       VAR4        VAR5
0      1      Prior             0.0000  24.390244   42.682927
1      2      Prior             6.0606  19.354839   25.806452
2      3      Prior            12.1212  35.087719  146.198830
3      4      Prior             3.3333  18.404908  110.429448
4      5      Prior            12.9032  17.964072   23.952096

In [28]:
treat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Month              20 non-null     int64  
 1   Reengineer         20 non-null     object 
 2   Employee_Turnover  20 non-null     float64
 3   VAR4               20 non-null     float64
 4   VAR5               20 non-null     float64
dtypes: float64(3), int64(1), object(1)
memory usage: 932.0+ bytes


In [29]:
treat.Reengineer.value_counts()

Reengineer
Prior    13
Post      7
Name: count, dtype: int64

In [30]:
#checking missisng values 
treat.isna().sum()

Month                0
Reengineer           0
Employee_Turnover    0
VAR4                 0
VAR5                 0
dtype: int64

### BUSINESS PROBLEM: Determine what effect, if any, the reengineering effort had on the 
### incidence behavioral problems and staff turnover. i.e To determine if the reengineering effort
### changed the critical incidence rate. Isthere evidence that the critical incidence rate
### improved?

In [31]:
#filtering reengineer data on the basis of employee turnover
prior=treat.Employee_Turnover[treat['Reengineer']=='Prior']
post=treat.Employee_Turnover[treat['Reengineer']=='Post']

In [32]:
u1=prior.mean()
u2=post.mean()
print('prior employee turnover mean:',u1)
print('post employee turnover mean:',u2)

prior employee turnover mean: 11.735692307692307
post employee turnover mean: 18.68912857142857


### 1.) Define Null Hypothesis (H0):
      #H0 : Employee turnover not improved after reengineer.
             u1>=u2    
### 2.) Define Alternate Hyothesis (Ha):
      #Ha : Employee turnover improved after reengineer.
             u1<u2
### 3.) Define CI and p-value:
      #CI = 95%
      #p-value=0.05%
### 4.) Perform Test:
      #T-Test

In [33]:
stats.ttest_ind(prior,post)

TtestResult(statistic=-1.770306704753604, pvalue=0.09361109345535304, df=18.0)

### Conclusion:
     # since the new p-value is 0.093>0.05 
     # hence we fail to reject the null hypothesis
    # therefore there is no evidence to prove the claim

In [34]:
#filtering reeinger data on the basis of var5(critical incidence)
prior1=treat.VAR5[treat['Reengineer']=='Prior']
post1=treat.VAR5[treat['Reengineer']=='Post']

In [35]:
u1=prior1.mean()
u2=post1.mean()
print('prior critical incidence mean:',u1)
print('post critical incidence mean:',u2)

prior critical incidence mean: 53.887906321846145
post critical incidence mean: 23.34971927988571


### 1.) Define Null Hypothesis (H0):
  #H0 : Critical incidence not improved after reengineer.
         u1>=u2    
### 2.) Define Alternate Hyothesis (Ha):
  #Ha : Critical incidence improved after reengineer.
         u1<u2
### 3.) Define CI and p-value:
  #CI = 95%
  #p-value=0.05%
### 4.) Perform Test:
  #T-Test

In [36]:
stats.ttest_ind(prior1,post1)

TtestResult(statistic=1.627914425352865, pvalue=0.12091989189884148, df=18.0)

### Conclusiion:
     # since the new p-value is 0.12>0.05
     # hence we fail to reject the null hypothesis
        #therefore there is no evidence to prove the claim

### BUSINESS PROBLEM - 4

In [37]:
PT=pd.read_csv('Priority_Assessment.csv')

In [38]:
PT.head()

Days Priority
0   3.3     High
1   7.9   Medium
2   0.3     High
3   0.7   Medium
4   8.6   Medium

In [39]:
PT.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 642 entries, 0 to 641
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Days      642 non-null    float64
 1   Priority  642 non-null    object 
dtypes: float64(1), object(1)
memory usage: 10.2+ KB


In [40]:
PT.isna().sum()

Days        0
Priority    0
dtype: int64

In [41]:
PT.Priority.value_counts()

Priority
High      326
Medium    249
Low        67
Name: count, dtype: int64

### BUSINESS PROBLEM: We will focus on the prioritization system. If the system is working, then
### high priority jobs, on average, should be completed more quickly than medium priority jobs,
### and medium priority jobs should be completed more quickly than low priority jobs. Use the
### data provided to determine whether thisis, in fact, occurring.

In [42]:
high=PT.Days[PT['Priority']=='High']
med=PT.Days[PT['Priority']=='Medium']
low=PT.Days[PT['Priority']=='Low']

In [43]:
u1=high.mean()
u2=med.mean()
u3=low.mean()
print('high priority mean:',u1)
print('medium priority mean:',u2)
print('low priority mean:',u3)

high priority mean: 3.0236196319018402
medium priority mean: 2.5
low priority mean: 4.228358208955224


### 1.) Define Null Hypothesis (H0):
#H0 : u1>=u2>=u3

### 2.) Define Alternate Hyothesis (Ha):
#Ha : u1<u2<u3

### 3.) Define CI and p-value:
#CI = 95% #p-value=0.05%

### 4.) Perform Test:
#F-Test

In [44]:
stats.f_oneway(high,med,low)

F_onewayResult(statistic=1.812311010076072, pvalue=0.16411459461716182)

### Conclusion:
     # since the new p-value is 0.16 > 0.05
    # hence we fail to reject the null hypothesis
    # there is no evidence for the claim

### BUSINESS PROBLEM - 5

In [45]:
film=pd.read_csv('Films.csv')

In [46]:
film.head()

_rowstate_                    Movie  Gender Marital_Status  Sinage  \
0           0  Ferris Buellers Day Off  Female        Married     2.0   
1           0  Ferris Buellers Day Off  Female         Single     1.0   
2           0  Ferris Buellers Day Off    Male        Married     2.0   
3           0  Ferris Buellers Day Off  Female        Married     1.0   
4           0  Ferris Buellers Day Off  Female        Married     1.0   

   Parking  Clean  Overall  Age  Income Hear_About  
0      2.0    2.0      2.0  3.0     1.0          5  
1      1.0    1.0      1.0  2.0     1.0          5  
2      4.0    3.0      2.0  4.0     1.0          5  
3      3.0    2.0      2.0  4.0     1.0          5  
4      1.0    1.0      1.0  3.0     3.0          1

In [47]:
film.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 330 entries, 0 to 329
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   _rowstate_      330 non-null    int64  
 1   Movie           330 non-null    object 
 2   Gender          330 non-null    object 
 3   Marital_Status  328 non-null    object 
 4   Sinage          328 non-null    float64
 5   Parking         328 non-null    float64
 6   Clean           327 non-null    float64
 7   Overall         328 non-null    float64
 8   Age             328 non-null    float64
 9   Income          314 non-null    float64
 10  Hear_About      323 non-null    object 
dtypes: float64(6), int64(1), object(4)
memory usage: 28.5+ KB


In [48]:
film.isna().sum()

_rowstate_         0
Movie              0
Gender             0
Marital_Status     2
Sinage             2
Parking            2
Clean              3
Overall            2
Age                2
Income            16
Hear_About         7
dtype: int64

In [49]:
film.Marital_Status=np.where(film.Marital_Status.isna(),film.Marital_Status.mode(),film.Marital_Status)
film.Sinage=np.where(film.Sinage.isna(),film.Sinage.mean(),film.Sinage)
film.Parking=np.where(film.Parking.isna(),film.Parking.mean(),film.Parking)
film.Clean=np.where(film.Clean.isna(),film.Clean.mean(),film.Clean)
film.Overall=np.where(film.Overall.isna(),film.Overall.mean(),film.Overall)
film.Age=np.where(film.Age.isna(),film.Age.mode(),film.Age)
film.Income=np.where(film.Income.isna(),film.Income.mode(),film.Income)
film.Hear_About=np.where(film.Hear_About.isna(),film.Hear_About.mode(),film.Hear_About)

In [50]:
film.isna().sum()

_rowstate_        0
Movie             0
Gender            0
Marital_Status    0
Sinage            0
Parking           0
Clean             0
Overall           0
Age               0
Income            0
Hear_About        0
dtype: int64

In [51]:
#data cleaning
film.Marital_Status.value_counts()

Marital_Status
2          210
1           88
Single      18
Married     12
Slngle       2
Name: count, dtype: int64

In [52]:
film['Marital_Status']=film.Marital_Status.replace('Married','married')
film['Marital_Status']=film.Marital_Status.replace('Single','single')
film['Marital_Status']=film.Marital_Status.replace('Slngle','single')
film['Marital_Status']=film.Marital_Status.replace('1','married')
film['Marital_Status']=film.Marital_Status.replace('2','single')


In [53]:
film.Marital_Status.value_counts()

Marital_Status
single     230
married    100
Name: count, dtype: int64

In [54]:
film.Gender.value_counts()

Gender
2         194
1         104
Female     19
Male       13
Name: count, dtype: int64

In [55]:
film['Gender']=film.Gender.replace('2','female')
film['Gender']=film.Gender.replace('1','male')
film['Gender']=film.Gender.replace('Female','female')
film['Gender']=film.Gender.replace('Male','male')

In [56]:
film.Gender.value_counts()

Gender
female    213
male      117
Name: count, dtype: int64

### BUSINESS PROBLEM: Use the survey resultsto addressthe following questions
###  What isthe overall level of customer satisfaction?

In [57]:
film.Overall.value_counts()

Overall
2.000000    160
1.000000    151
3.000000     12
5.000000      4
1.618902      2
4.000000      1
Name: count, dtype: int64

In [58]:
film.Overall.mean()

1.6189024390243902

In [59]:
print('Therefore the overall satisfaction rating is 2, and the overall mean rating is',film.Overall.mean())

Therefore the overall satisfaction rating is 2, and the overall mean rating is 1.6189024390243902


###  What factors are linked to satisfaction?

In [60]:
# observed frequency table 
tab1=pd.crosstab(film.Overall,film.Age)
tab2=pd.crosstab(film.Overall,film.Gender)
tab3=pd.crosstab(film.Overall,film.Marital_Status)
tab4=pd.crosstab(film.Overall,film.Movie)
tab5=pd.crosstab(film.Overall,film.Income)
tab6=pd.crosstab(film.Overall,film.Clean)
tab7=pd.crosstab(film.Overall,film.Parking)
tab8=pd.crosstab(film.Overall,film.Sinage)


### 1.) Define Null Hypothesis (H0):
#H0 : All the factors are effecting overall satisfaction.

### 2.) Define Alternate Hyothesis (Ha):
#Ha : All the factors are not effecting overall satisfaction.

### 3.) Define CI and p-value:
#CI = 95% #p-value=0.05%

### 4.) Perform Test:
#Chi-square test

In [61]:
stats.chi2_contingency(tab1)

Chi2ContingencyResult(statistic=12.138966663644657, pvalue=0.6684813226405949, dof=15, expected_freq=array([[1.18969697e+01, 8.09909091e+01, 5.35363636e+01, 4.57575758e+00],
       [1.57575758e-01, 1.07272727e+00, 7.09090909e-01, 6.06060606e-02],
       [1.26060606e+01, 8.58181818e+01, 5.67272727e+01, 4.84848485e+00],
       [9.45454545e-01, 6.43636364e+00, 4.25454545e+00, 3.63636364e-01],
       [7.87878788e-02, 5.36363636e-01, 3.54545455e-01, 3.03030303e-02],
       [3.15151515e-01, 2.14545455e+00, 1.41818182e+00, 1.21212121e-01]]))

In [62]:
stats.chi2_contingency(tab2)

Chi2ContingencyResult(statistic=4.488568645130532, pvalue=0.48141436969705187, dof=5, expected_freq=array([[ 97.46363636,  53.53636364],
       [  1.29090909,   0.70909091],
       [103.27272727,  56.72727273],
       [  7.74545455,   4.25454545],
       [  0.64545455,   0.35454545],
       [  2.58181818,   1.41818182]]))

In [63]:
stats.chi2_contingency(tab3)

Chi2ContingencyResult(statistic=5.532821048085229, pvalue=0.35436035505007557, dof=5, expected_freq=array([[ 45.75757576, 105.24242424],
       [  0.60606061,   1.39393939],
       [ 48.48484848, 111.51515152],
       [  3.63636364,   8.36363636],
       [  0.3030303 ,   0.6969697 ],
       [  1.21212121,   2.78787879]]))

In [64]:
stats.chi2_contingency(tab4)

Chi2ContingencyResult(statistic=12.511868043591603, pvalue=0.2522577891412043, dof=10, expected_freq=array([[62.68787879, 14.64242424, 73.66969697],
       [ 0.83030303,  0.19393939,  0.97575758],
       [66.42424242, 15.51515152, 78.06060606],
       [ 4.98181818,  1.16363636,  5.85454545],
       [ 0.41515152,  0.0969697 ,  0.48787879],
       [ 1.66060606,  0.38787879,  1.95151515]]))

In [65]:
stats.chi2_contingency(tab5)

Chi2ContingencyResult(statistic=7.418499723181489, pvalue=0.6854328680629385, dof=10, expected_freq=array([[72.2969697 , 37.52121212, 41.18181818],
       [ 0.95757576,  0.4969697 ,  0.54545455],
       [76.60606061, 39.75757576, 43.63636364],
       [ 5.74545455,  2.98181818,  3.27272727],
       [ 0.47878788,  0.24848485,  0.27272727],
       [ 1.91515152,  0.99393939,  1.09090909]]))

In [66]:
stats.chi2_contingency(tab6)

Chi2ContingencyResult(statistic=345.7963259374637, pvalue=3.462183654474607e-58, dof=25, expected_freq=array([[6.58909091e+01, 1.37272727e+00, 6.95515152e+01, 1.09818182e+01,
        2.28787879e+00, 9.15151515e-01],
       [8.72727273e-01, 1.81818182e-02, 9.21212121e-01, 1.45454545e-01,
        3.03030303e-02, 1.21212121e-02],
       [6.98181818e+01, 1.45454545e+00, 7.36969697e+01, 1.16363636e+01,
        2.42424242e+00, 9.69696970e-01],
       [5.23636364e+00, 1.09090909e-01, 5.52727273e+00, 8.72727273e-01,
        1.81818182e-01, 7.27272727e-02],
       [4.36363636e-01, 9.09090909e-03, 4.60606061e-01, 7.27272727e-02,
        1.51515152e-02, 6.06060606e-03],
       [1.74545455e+00, 3.63636364e-02, 1.84242424e+00, 2.90909091e-01,
        6.06060606e-02, 2.42424242e-02]]))

In [67]:
stats.chi2_contingency(tab7)

Chi2ContingencyResult(statistic=313.6470148499061, pvalue=1.0870837819160887e-51, dof=25, expected_freq=array([[6.17727273e+01, 9.15151515e-01, 7.00090909e+01, 1.32696970e+01,
        3.20303030e+00, 1.83030303e+00],
       [8.18181818e-01, 1.21212121e-02, 9.27272727e-01, 1.75757576e-01,
        4.24242424e-02, 2.42424242e-02],
       [6.54545455e+01, 9.69696970e-01, 7.41818182e+01, 1.40606061e+01,
        3.39393939e+00, 1.93939394e+00],
       [4.90909091e+00, 7.27272727e-02, 5.56363636e+00, 1.05454545e+00,
        2.54545455e-01, 1.45454545e-01],
       [4.09090909e-01, 6.06060606e-03, 4.63636364e-01, 8.78787879e-02,
        2.12121212e-02, 1.21212121e-02],
       [1.63636364e+00, 2.42424242e-02, 1.85454545e+00, 3.51515152e-01,
        8.48484848e-02, 4.84848485e-02]]))

In [68]:
stats.chi2_contingency(tab8)

Chi2ContingencyResult(statistic=208.15634074943048, pvalue=8.18689961410127e-31, dof=25, expected_freq=array([[5.17060606e+01, 9.15151515e-01, 7.13818182e+01, 2.19636364e+01,
        2.74545455e+00, 2.28787879e+00],
       [6.84848485e-01, 1.21212121e-02, 9.45454545e-01, 2.90909091e-01,
        3.63636364e-02, 3.03030303e-02],
       [5.47878788e+01, 9.69696970e-01, 7.56363636e+01, 2.32727273e+01,
        2.90909091e+00, 2.42424242e+00],
       [4.10909091e+00, 7.27272727e-02, 5.67272727e+00, 1.74545455e+00,
        2.18181818e-01, 1.81818182e-01],
       [3.42424242e-01, 6.06060606e-03, 4.72727273e-01, 1.45454545e-01,
        1.81818182e-02, 1.51515152e-02],
       [1.36969697e+00, 2.42424242e-02, 1.89090909e+00, 5.81818182e-01,
        7.27272727e-02, 6.06060606e-02]]))

### Conclusion:
     # so the new p-value of tab 6,7 and 8 < 0.05 
    # hence we reject the null hypothesis
    # therefore clean, parking and sinage is affecting overall satisfaction


###  What is the demographic profile of Film on the Rocks patrons?

In [69]:
#ratio of married and single people
film.Marital_Status.value_counts()/film.Marital_Status.count()

Marital_Status
single     0.69697
married    0.30303
Name: count, dtype: float64

In [70]:
#ratio of male and female
film.Gender.value_counts()/film.Gender.count()

Gender
female    0.645455
male      0.354545
Name: count, dtype: float64

In [71]:
#ratio of different age groups
film.Age.value_counts()/film.Age.count()

Age
2.0    0.536364
3.0    0.354545
1.0    0.078788
4.0    0.030303
Name: count, dtype: float64

In [72]:
#ratio of different income groups
film.Income.value_counts()/film.Income.count()

Income
1.0    0.478788
3.0    0.272727
2.0    0.248485
Name: count, dtype: float64

###  In what media outlet(s) should the film series be advertised?

In [73]:
film.Hear_About.value_counts()
# Hear about is the best way to advertise

Hear_About
5      233
4       41
1       22
3       14
2       12
2,5      2
3,4      2
4,5      1
1,5      1
5,4      1
3,5      1
Name: count, dtype: int64